pkg/apis/infinispan/v1/infinispan_types.go

```diff
@@ -191,17 +191,28 @@ type InfinispanExternalDependencies struct {
 	VolumeClaimName string `json:"volumeClaimName,optional,omitempty"`
 }
 
+// InfinispanCloudEvents describes how Infinispan is connected with Cloud Event, see Kafka docs for more info
+type InfinispanCloudEvents struct {
+	// BootstrapServers is comma separated list of boostrap server:port addresses
+	BootstrapServers string `json:"bootstrapServers,optional,omitempty"`
+	// Acks configuration for the producer ack-value
+	Acks string `json:"acks,optional,omitempty"`
+	// CacheEntriesTopic is the name of the topic on which events will be published
+	CacheEntriesTopic string `json:"cacheEntriesTopic,optional,omitempty"`
+}
+
 // InfinispanSpec defines the desired state of Infinispan
 type InfinispanSpec struct {
-	Replicas  int32                   `json:"replicas"`
-	Image     *string                 `json:"image,optional,omitempty"`
-	Security  InfinispanSecurity      `json:"security,optional,omitempty"`
-	Container InfinispanContainerSpec `json:"container,optional,omitempty"`
-	Service   InfinispanServiceSpec   `json:"service,optional,omitempty"`
-	Logging   *InfinispanLoggingSpec  `json:"logging,optional,omitempty"`
-	Expose    *ExposeSpec             `json:"expose,optional,omitempty"`
-	Autoscale *Autoscale              `json:"autoscale,optional,omitempty"`
-	Affinity  *corev1.Affinity        `json:"affinity,optional,omitempty"`
+	Replicas    int32                   `json:"replicas"`
+	Image       *string                 `json:"image,optional,omitempty"`
+	Security    InfinispanSecurity      `json:"security,optional,omitempty"`
+	Container   InfinispanContainerSpec `json:"container,optional,omitempty"`
+	Service     InfinispanServiceSpec   `json:"service,optional,omitempty"`
+	Logging     *InfinispanLoggingSpec  `json:"logging,optional,omitempty"`
+	Expose      *ExposeSpec             `json:"expose,optional,omitempty"`
+	Autoscale   *Autoscale              `json:"autoscale,optional,omitempty"`
+	Affinity    *corev1.Affinity        `json:"affinity,optional,omitempty"`
+	CloudEvents *InfinispanCloudEvents  `json:"cloudEvents,optional"`
```

Comment:
@rigazilla `optional` is incorrect JSON directive. Please see my golangci-lint PR. It would be nice to merge it before, to avoid many potential golang code issues in the future.

Reply:
done